Docs: update API access for new token type (#1958)
Unleash/unleash: website/docs/sdks/proxy-javascript.md

title: JavaScript Proxy SDK
---

> In this guide we explain how to use feature toggles in a Single Page App via [Unleash Proxy](/sdks/unleash-proxy) or directly to Unleash (since _v4.16_). You can also checkout the source code for the [JavaScript Proxy SDK](https://github.com/unleash/unleash-proxy-client-js).

Reviewer: ```suggestion
This guide shows you how to use feature toggles in a single-page app with the [Unleash Proxy](/sdks/unleash-proxy) and the [direct access API](../reference/frontend-api). You can also check out the source code for the [JavaScript Proxy SDK](https://github.com/unleash/unleash-proxy-client-js) on GitHub for more details around the SDK.
```

## Introduction {#introduction}

> For single-page apps we have a tiny proxy-client in JavaScript, without any external dependencies, except from browser APIs. This client will store toggles relevant for the current user in local-storage and synchronize with the Unleash in the background. This means we can bootstrap the toggles for a specific use the next time the user visits the web-page.

Reviewer: ```suggestion
The JavaScript proxy client is a tiny Unleash client written in JavaScript without any external dependencies (except from browser APIs). This client stores toggles relevant for the current user in `localStorage` and synchronizes with Unleash (the proxy _or_ the direct access API) in the background. Because toggles are stored in the user's browser, the client  can use them to bootstrap itself the next time the user visits the same web page.
```

> We are looking in to also [supporting react-native](https://github.com/Unleash/unleash/issues/785) with this SDK. Reach out if you want to help us validate the implementation.

## How to use the JavaScript Proxy SDK

> ### Step 1: Install

Reviewer: ```suggestion
## Step 1: Install
```

```shell npm2yarn
npm install unleash-proxy-client
```

> ### Step 2: Initialize the SDK

Reviewer: ```suggestion
## Step 2: Initialize the SDK
```

```js
import { UnleashClient } from 'unleash-proxy-client';

const unleash = new UnleashClient({
  url: 'https://eu.unleash-hosted.com/hosted/proxy',

>   clientKey: 'your-secret-key',

Reviewer: Why is this key suddenly secret? That seems strange.

  appName: 'my-webapp',
});

> // Used to set the context fields, shared with Unleash Proxy

Reviewer: ```suggestion
// Use `updateContext` to set Unleash context fields.
```

unleash.updateContext({ userId: '1233' });

// Start the background polling
unleash.start();
```

> #### Option A: With Unleash Proxy

Reviewer: ```suggestion
### Option A: Connecting to the Unleash proxy
```

> You need to have an Unleash Proxy server running. Fill `url` and a proxy-specific `clientKey` in order to connect. For more on how to set up client keys, [consult the Unleash Proxy docs](unleash-proxy.md#configuration-variables).

Reviewer: ```suggestion
:::tip Prerequisites

To connect to an Unleash proxy, you need to have an instance of the proxy running. 

:::

Add the proxy's URL and a [proxy client key](../reference/api-tokens-and-client-keys.mdx#proxy-client-keys). 
The [_configuration_ section of the Unleash proxy docs](unleash-proxy.md#configuration-variables) contain more info on how to configure client keys for your proxy.
```

> #### Option B: Directly to Unleash (since _Unleash v4.16_)

Reviewer: ```suggestion
### Option B: Connecting directly to Unleash
```

> In this case `url` will be `https://<YOUR_INSTANCE_URL>/api/frontend` and you can generate `FRONTEND` token in the UI - see [How to create API token](/user_guide/api-token). You might also need to set up CORS for your instance.

Reviewer: ```suggestion
Use the url to your Unleash instance's direct access API (<unleash-url>/api/frontend) as the `url` parameter. For the `clientKey` parameter, use a `FRONTEND` token generated from your Unleash instance. Refer to the 
[_how to create API tokens_](/user_guide/api-token) guide for the necessary steps. 

You might also need to set up cross-origin resource sharing (CORS) for your instance. Visit the [CORS section of the direct access API guide](../reference/frontend-api.mdx#cors) for more information on setting up CORS.
```

}
```

> ### Listen for updates via the EventEmitter

Reviewer: ```suggestion
## Listen for updates via the EventEmitter
```
